lms/djangoapps/courseware/tabs.py

```diff
@@ -66,8 +68,7 @@ def is_enabled(cls, course, user=None):
         """
         The "Home" tab is not shown for the new unified course experience.
         """
-        request = RequestCache.get_current_request()
-        return not waffle.flag_is_active(request, UNIFIED_COURSE_EXPERIENCE_FLAG)
+        return not course_experience_config().is_enabled(UNIFIED_COURSE_EXPERIENCE_FLAG, course.id)
```

Comment:
@andy-armstrong Thanks for all the great feedback.  I'm glad I finally got it in shape enough for someone to see it.  My head was in the weeds, and I needed some outside perspective.

I like your idea around `UNIFIED_COURSE_EXPERIENCE_FEATURE_FLAG`.  It seems a little unexpected, but it also seems quite clean and reads well.  I think I should play with that a bit.